proxy/src/control/destination/background.rs

```diff
@@ -581,18 +584,22 @@ fn pb_to_addr_meta(
         .collect::<Vec<_>>();
     labels.sort_by(|(k0, _), (k1, _)| k0.cmp(k1));
 
-    let tls_identity = pb.tls_identity.and_then(|pb| {
+    let mut tls_identity =
+        Conditional::None(tls::ReasonForNoIdentity::NotProvidedByServiceDiscovery);
+    if let Some(pb) = pb.tls_identity {
```

Comment:
tioli: i'd probably write this as `let tls_identity = match pb.tls_identity { ... };` rather than use a `mut`

Reply:
That's normally how I would do it too, but in this case the `NotProvidedByServiceDiscovery` would be repeated three times.